lib/facility.rb

```diff
@@ -1,14 +1,81 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
+              
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(info_hash, services = [])
+    @name = info_hash[:name]
+    @address = info_hash[:address]
+    @phone = info_hash[:phone]
+    @services = services
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless @services.include?('Vehicle Registration')
+    if vehicle.antique?
```

Comment:
Refactor Opportunity (for the future, we know time is limited!!): Consider, when you have a long method like this, how you can use helper methods to break up the logic. For example, you could have a helper method here that you call `process_antique_registration` and it can handle setting the plate type, setting registration date, collecting the right amount of fees for this type of car, etc.

Using helper methods can help a lot when we have complex or nested conditionals, because those get hard to read. If you call helper methods though, you're able to extract some of that logic into a separate location and give it a readable name to indicate to another dev what is happening.